drivers/io/io_fip.c

```diff
@@ -272,7 +272,9 @@ static int fip_file_read(io_entity_t *entity, uintptr_t buffer, size_t length,
 	uintptr_t backend_handle;
 
 	assert(entity != NULL);
+#ifndef PLAT_ALLOW_ZERO_ADDR_COPY
 	assert(buffer != (uintptr_t)NULL);
+#endif
```

Comment:
This is the first platform that needs load an image at 0. It makes invalid the asserts, and since this code is going to be executed with the MMU enabled, the assert doesn't help too much. I think it is better if you remove the NULL checks in these asserts.

Reply:
@robertovargas-arm, Thank you! Should I remove all cases where the buffer pointer is checked for 0? In other words - should I remove the PLAT_ALLOW_ZERO_ADDR_COPY definition along with every call to "assert(buffer != (uintptr_t)NULL)"?

Reply:
Yes. We can see now that this check is invalid in one platform, so it doesn't make sense. As the MMU is already enabled, if 0 is not a valid address then it will generate a fault abort, so the assert is not adding too much. I see a bit dangerous that you load something using the address 0, because it means that any NULL pointer will not be caught, opening a door to silent errors that can be exposed later, but this is a platform decision.